actionpack/lib/action_dispatch/routing/route_set.rb

```diff
@@ -292,7 +292,15 @@ def define_url_helper(mod, route, name, opts, route_key, url_strategy)
           mod.module_eval do
             define_method(name) do |*args|
               options = nil
-              options = args.pop if args.last.is_a? Hash
+
+              if args.last.is_a? Hash
+                options = args.pop 
+              end
+
+              if options.is_a? ActiveSupport::HashWithIndifferentAccess
+                options = options.symbolize_keys 
```

Comment:
We should not call `symbolize_keys` in any case. I think the only good options is to remove the deprecation and change all code to only work with string keys, not symbol keys.